src/Device/Driver/XCVario.cpp

```diff
@@ -42,28 +47,42 @@ class XVCDevice : public AbstractDevice {
 
   // virtual methods from class Device
   bool ParseNMEA(const char *line, struct NMEAInfo &info) override;
-
   bool PutMacCready(double mc, OperationEnvironment &env) override;
   bool PutBugs(double bugs, OperationEnvironment &env) override;
   bool PutBallast(double fraction, double overload, OperationEnvironment &env) override;
   bool PutQNH(const AtmosphericPressure &pres, OperationEnvironment &env) override;
+  bool EnableNMEA(OperationEnvironment &env) override;
+  void OnSysTicker() override;
+  void LinkTimeout() override;
+
   bool PutCrewWeight(double weight, OperationEnvironment &env) override;
   bool PutEmptyMass(double weight, OperationEnvironment &env) override;
   bool SentNMEAItem( const char *which, int value, OperationEnvironment &env );
+
+
+private:
+  int protocol_version = XCV_VERSION_UNKNOWN;
+  int link_timer = 5;
```

Comment:
This looks like you implemented a timeout detection inside this driver, but there is already `OnLinkTimeout()`, which you also use. Don't add unnecessary complexity to the driver.

Reply:
its actually an application layer XCV protocol supervision and on top of the LinkTimeout()

Reply:
I don't understand. This needs further explanation.

Reply:
Once the link comes up, XCVario protocol needs further negotiation on application layer what protocol to run. To be backward compatible with older device SW, there is protocol number negotiation between XCS and the XCV device, and until this hasn't been finished, we can't clearly go into full operation, so there is a second decision on top of the link status below on application layer to decide about this. Older devices might not answer the protocol version query, so after the link is up and if within a timeout of 5 seconds there is no answer, XCS will run in legacy mode with fractional ballasts. And until then, as long as the protocol version is unknown, any sending of NMEA except protocol version negotiation, must not take place.